test/framework/src/main/java/org/elasticsearch/cluster/ESAllocationTestCase.java

```diff
@@ -55,6 +58,14 @@
     private static final ClusterSettings EMPTY_CLUSTER_SETTINGS =
         new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
 
+    public static final SnapshotsInfoService SNAPSHOT_INFO_SERVICE_WITH_SHARD_SIZES = () ->
+        new SnapshotShardSizeInfo(ImmutableOpenMap.of()) {
+            @Override
+            public Long getShardSize(ShardRouting shardRouting) {
+                return randomNonNegativeLong();
```

Comment:
Should we make sure to return the same size for the same shard consistently? (and maybe also assert that we are indeed dealing with a snapshot recovery routing here?)

Reply:
I think that it is trappy to always return an random size here. Test that rely on snapshot shard information should use their own implementation. I pushed 28de2d67de1b28c7d3e29e46bda50e2bef38baa9 to add an assertion as you suggested + throw an unsupported exception here.